app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -213,25 +218,42 @@ export default Component.extend(
       });
     },
 
+    _destroyUserStatusInstances(instances) {
+      instances.forEach((instance) => {
+        instance.destroy();
+      });
+      instances = [];
+    },
+
     @on("didInsertElement")
     _composerEditorInit() {
       const $input = $(this.element.querySelector(".d-editor-input"));
 
       if (this.siteSettings.enable_mentions) {
         $input.autocomplete({
           template: findRawTemplate("user-selector-autocomplete"),
-          dataSource: (term) =>
-            userSearch({
+          dataSource: (term) => {
+            this._destroyUserStatusInstances(this.userStatusInstances);
```

Comment:
I think the previous implementation, when we didn't pass `this.userStatusInstances` to the function was cleaner. Looks like we still can do it this way:

```js
this._destroyUserStatusInstances();
```